### PR TITLE
Verify selected layer bounds

### PR DESCRIPTION
**Pull request by iwehrman**
This PR adds a post-verification step for selected layer bounds to actions that modify the bounds. It also "cleans up" (and hopefully doesn't break?) the layer transform event handler.

With this on, I immediately notice a few warnings. Some of them are "false" in that there are actions racing to update the bounds, and the post-verification step sometimes runs before a later action fixes things up. I also see a number of what appear to be true (if perhaps not severe?) warnings when transforming layers via the transform panel due to rounding issues. In any case, I think we should add these warnings first and triage them later.

### CODE REVIEW

File: src/js/actions/transform.js [hunk 1]
```diff
@@ -1241,7 +1250,71 @@ define(function (require, exports) {
...
+
+        var appStore = this.flux.store("application"),
+            currentDoc = appStore.getCurrentDocument();
+
+        if (!currentDoc) {
+            return Promise.resolve();
+        }
+
+        // newDuplicateSheets move events should be processed immediately, not debounced
+        if (event.newDuplicateSheets) {
```
Comment (iwehrman) [inline src/js/actions/transform.js]: This used to be debounced, but the debouncing was broken in the sense that it was functionally identical to just delaying the code from executing by 200ms. I couldn't think of any reason why that delay was necessary either, so I just removed it. It's entirely possible that this breaks something, but at least it's less mysterious...

File: src/js/models/layerstructure.js [hunk 2]
```diff
@@ -464,7 +464,7 @@ define(function (require, exports, module) {
         /**
          * The subset of Layer models that are all selected or are descendants of selected
          *
-         * @return {Immutable.List.<Layer>}
+         * @return {Immutable.OrderedSet.<Layer>}
```
Comment (iwehrman) [inline src/js/models/layerstructure.js]: Unrelated JSDoc fix.

File: src/js/actions/transform.js (hunk 1 above)
Comment (volfied) [inline src/js/actions/transform.js]: We'll see!

**Comment by volfied** [comment]
Nice, thanks for doing this.
